python/TestHarness/TestHarness.py

```diff
@@ -362,11 +362,8 @@ def canPrint(self, tester):
 
     # Format the status message to make any caveats easy to read when they are printed
     def formatCaveats(self, tester):
-        caveats = []
         result = ''
-
-        if tester.specs.isValid('caveats'):
-            caveats = tester.specs['caveats']
+        caveats = list(tester.getCaveats())
```

Comment:
Hmm, I suppose we can allow this but you really should follow up and try to clean this up everywhere. I'm not a huge fan of testing against python types to do different things. I'll comment on that below.